Automattic/jetpack
# Recurring Payments: Open new window to connect instead of loading in place

…and connection flows in the same window.




Partially fixes https://github.com/Automattic/view-design/issues/275

#### Changes proposed in this Pull Request:

* When inside the widget editor or customizer view we do not want to open links within the same window. This will cause issues with losing design changes. This PR will check to see if we are in the customizer or widget editor and open a new window instead, where the flow can be completed.

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Go to the customizer using the WordPress 5.8 branch on WordPress.com (use `svn switch` and ping me for the branch URL)
* On a sandboxed site go to the customizer and then to widgets > [widget area]
* Insert a payments block
* Click on the "Connect Stripe" option in the block toolbar.
* Confirm this opens a new window.
* Do the same in the post and site editors and confirm it still loads in the same window.
